tools/configure/configure.ml

```diff
@@ -518,23 +67,29 @@ let _ =
 
 (** * Browser command *)
 
-let browser =
+let browser prefs arch =
   match prefs.browser with
   | Some b -> b
-  | None when arch_is_win32 -> "start %s"
+  | None when arch_is_win32 arch -> "start %s"
   | None when arch = "Darwin" -> "open %s"
   | _ -> "firefox -remote \"OpenURL(%s,new-tab)\" || firefox %s &"
 
 (** * OCaml programs *)
+module CamlConf = struct
+  type t =
```

Comment:
Perhaps this is a bit off topic, but why do you prefer "Haskell style" records here? I believe that keeping the semicolon on the line it is ending is much clearer.

Reply:
This style is also widely used in OCaml, for example in Dune and Jane street libs I recon.

Reply:
Fair enough, like I said it's probably a discussion for another time.